WolvenKit.App/Services/UpdateService.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Semver;
+using WolvenKit.App.Helpers;
+using WolvenKit.App.Models;
+using WolvenKit.Common;
+using WolvenKit.Core.Interfaces;
+using System.Text.Json;
+using Octokit;
+
+namespace WolvenKit.App.Services;
+
+public class UpdateService : IUpdateService
+{
+    private readonly ILoggerService _loggerService;
+    private readonly ISettingsManager _settingsManager;
+    private readonly HttpClient _httpClient;
+    private readonly GitHubClient _gitHubClient;
+    
+    public UpdateService(ILoggerService loggerService, ISettingsManager settingsManager)
+    {
+        _loggerService = loggerService;
+        _settingsManager = settingsManager;
+        _httpClient = new HttpClient();
+        _httpClient.DefaultRequestHeaders.Add("User-Agent", "WolvenKit");
+        _gitHubClient = new GitHubClient(new ProductHeaderValue("WolvenKit"));
+    }
+    
+    public async Task UpdateToNewestVersion()
+    {
+        if (!await IsUpdateAvailable())
+        {
+            return;
+        }
+        
+        var latestRelease = await GetLatestRelease();
+        if (latestRelease is null)
+        {
+            return;
+        }
+        
+        var downloadZipPath = Path.Join(ISettingsManager.GetTemp_DownloadsPath(), latestRelease.TagName + ".zip");
+        var portableAsset = latestRelease.Assets.First(a => a.Name == $"WolvenKit-{latestRelease.TagName}.zip");
+        
+        try
+        {
+            var responseAsset = await _httpClient.GetAsync(portableAsset.BrowserDownloadUrl);
+            responseAsset.EnsureSuccessStatusCode();
+            await File.WriteAllBytesAsync(downloadZipPath, await responseAsset.Content.ReadAsByteArrayAsync());
+        }
+        catch (HttpRequestException ex)
+        {
+            _loggerService.Error($"Failed to download update: {portableAsset.BrowserDownloadUrl}");
+            _loggerService.Error(ex);
+            return;
+        }
+        
+        var unzipPath = Path.Join(ISettingsManager.GetTemp_DownloadsPath(), "unzip");
+        Directory.CreateDirectory(unzipPath);
+        ZipFile.ExtractToDirectory(downloadZipPath, unzipPath);
+        
+        var exePath = Environment.ProcessPath ?? Path.Combine(AppContext.BaseDirectory, "WolvenKit.exe");
+        var scriptPath = Path.Combine(ISettingsManager.GetTemp_DownloadsPath(), "update.ps1");
```

Comment:
Why do we need both a Powershell and VBS to perform the update? I understand offloading the download and update to a separate process so you can access the locked files, but why both?